--- a/CMakeLists.txt
+++ a/CMakeLists.txt
@@ -100,6 +100,8 @@
 option(POCO_UNBUNDLED
   "Set to OFF|ON (default is OFF) to control linking dependencies as external" OFF)
 
+option(ARCH_PPC "Are we compiling for PPC architecture? Default is OFF" OFF)
+
 # Uncomment from next two lines to force statitc or dynamic library, default is autodetection
 if(POCO_STATIC)
     add_definitions( -DPOCO_STATIC -DPOCO_NO_AUTOMATIC_LIBS)
@@ -128,6 +130,11 @@
 include(CheckTypeSize)
 find_package(Cygwin)
 
+# We are compiling for Turris PPC environment
+if(ARCH_PPC)
+  add_definitions( -DPOCO_NO_SHAREDMEMORY -DPOCO_NO_WSTRING -DPOCO_NO_FPENVIRONMENT)
+endif ()
+
 # OS Detection
 if(WIN32)
   add_definitions( -DPOCO_OS_FAMILY_WINDOWS -DUNICODE -D_UNICODE)
